conjure_oxide/tests/custom_tests.rs

```diff
@@ -0,0 +1,84 @@
+use pretty_assertions::assert_eq;
+use std::env;
+use std::error::Error;
+use std::fs;
+use std::path::PathBuf;
+use std::process::{Command, Output};
+
+pub fn custom_test(test_dir: &str) -> Result<(), Box<dyn Error>> {
+    let accept = env::var("ACCEPT").unwrap_or("false".to_string()) == "true";
+
+    // Convert test directory to a PathBuf
+    let test_path = PathBuf::from(test_dir);
+    assert!(
+        test_path.exists(),
+        "Test directory not found: {:?}",
+        test_path
+    );
+
+    // Get paths
+    let script_path = test_path.join("run.sh");
+    assert!(
+        script_path.exists(),
+        "Test script not found: {:?}",
+        script_path
+    );
+    let expected_output_path = test_path.join("stdout.expected");
+    let expected_error_path = test_path.join("stderr.expected");
+
+    // Locate `conjure_oxide` automatically in target/debug or target/release
+    let mut conjure_oxide_path = env::current_exe().unwrap();
+    conjure_oxide_path.pop();
+    conjure_oxide_path.pop();
+    conjure_oxide_path.push("conjure_oxide");
+
+    // Modify PATH so run.sh can find conjure_oxide
+    let mut path_var = env::var("PATH").unwrap_or_else(|_| "".to_string());
+    let conjure_dir = conjure_oxide_path.parent().unwrap().to_str().unwrap();
+    path_var = format!("{}:{}", conjure_dir, path_var);
+
+    // Execute the test script in the correct directory
+    let output: Output = Command::new("sh")
+        .arg("run.sh")
+        .current_dir(&test_path)
+        .env("PATH", path_var)
+        .output()?;
+
+    // Convert captured output/error to string
+    let actual_output = String::from_utf8_lossy(&output.stdout);
+    let actual_error = String::from_utf8_lossy(&output.stderr);
+
+    if accept {
+        // Overwrite expected files
+        if !actual_output.trim().is_empty() {
```

Comment:
I suggest adding an else case to this if (and the following). in the else case remove the file if it exists.

this is becaause: if the file had some content before, but now it should be updated to be empty, we leave it as is at the moment. does that make sense?

also, you can extract this logic to a new function called update_file which is called twice here, once for stdout and once for stderr. to reduce code duplication.